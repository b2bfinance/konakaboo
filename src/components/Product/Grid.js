import React from 'react';
import styled from 'styled-components';
import Typography from '@material-ui/core/Typography';
import Col from '../Col';
import { queries } from '../../utils/media';

export const Wrapper = styled(Col)`
  border-top: 2px solid ${props => props.theme.productBorder};
  padding: 16px 8px;
  text-align: center;

  ${queries.tablet`
    margin-top: 16px;
    margin-bottom: 16px;
    padding: 8px;
    border-top: 0;
    border-left: 2px solid ${props => props.theme.productBorder};
  `};

  &:first-child {
    border-top: 0;
    border-left: 0;
  }
`;

export const ProductMultiGrid = ({ columns, theme }) =>
  columns.map((column, i) => (
    <Wrapper
      key={i}
      background={(i + 1) % 2}
      theme={theme}
      item
      phone={100}
      tablet={80 / columns.length}
    >
      <Typography variant="subtitle2" component="span">
        {column.label}
      </Typography>
      <Typography variant="h5" component="span" color="textSecondary">
        {column.value}
      </Typography>
      {column.subtext && (
        <Typography variant="caption">{column.subtext}</Typography>
      )}
    </Wrapper>
  ));

export default Wrapper;
