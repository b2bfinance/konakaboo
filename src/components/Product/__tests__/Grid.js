import React from 'react';
import { render } from 'enzyme';
import Grid, { ProductMultiGrid } from '../Grid';

describe('Grid', () => {
  test('renders correctly', () => {
    expect(render(<Grid theme={stubData.theme} />)).toMatchSnapshot();
  });
});

describe('ProductMultiGrid', () => {
  test('renders correctly', () => {
    const columns = [
      { label: 'foo', value: 'bar' },
      { label: 'bar', value: 'foo' }
    ];

    expect(
      render(<ProductMultiGrid columns={columns} theme={stubData.theme} />)
    ).toMatchSnapshot();
  });
});
