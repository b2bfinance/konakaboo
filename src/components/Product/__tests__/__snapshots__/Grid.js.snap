// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
.c0 {
  min-height: 1px;
  border-top: 2px solid #eceff1;
  padding: 16px 8px;
  text-align: center;
}

.c0:first-child {
  border-top: 0;
  border-left: 0;
}

@media (min-width:36em) {
  .c0 {
    margin: 8px 0;
    padding: 8px 16px;
    border-top: 0;
    border-left: 2px solid #eceff1;
  }
}

<div
  class="c0"
/>
`;

exports[`ProductMultiGrid renders correctly 1`] = `
Array [
  .c0 {
  min-height: 1px;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
  border-top: 2px solid #eceff1;
  padding: 16px 8px;
  text-align: center;
}

.c0:first-child {
  border-top: 0;
  border-left: 0;
}

@media (min-width:36em) {
  .c0 {
    -webkit-flex: 0 0 40%;
    -ms-flex: 0 0 40%;
    flex: 0 0 40%;
    max-width: 40%;
  }
}

@media (min-width:36em) {
  .c0 {
    margin: 8px 0;
    padding: 8px 16px;
    border-top: 0;
    border-left: 2px solid #eceff1;
  }
}

<div
    class="c0"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2"
    >
      foo
    </h6>
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
    >
      bar
    </h5>
  </div>,
  .c0 {
  min-height: 1px;
  -webkit-flex: 0 0 100%;
  -ms-flex: 0 0 100%;
  flex: 0 0 100%;
  max-width: 100%;
  border-top: 2px solid #eceff1;
  padding: 16px 8px;
  text-align: center;
}

.c0:first-child {
  border-top: 0;
  border-left: 0;
}

@media (min-width:36em) {
  .c0 {
    -webkit-flex: 0 0 40%;
    -ms-flex: 0 0 40%;
    flex: 0 0 40%;
    max-width: 40%;
  }
}

@media (min-width:36em) {
  .c0 {
    margin: 8px 0;
    padding: 8px 16px;
    border-top: 0;
    border-left: 2px solid #eceff1;
  }
}

<div
    class="c0"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2"
    >
      bar
    </h6>
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
    >
      foo
    </h5>
  </div>,
]
`;
